src/python.cc

```diff
@@ -361,8 +421,394 @@ class ModelInstanceState : public BackendModelInstance {
       std::vector<std::unique_ptr<InferRequest>>& pb_inference_requests,
       AllocatedSharedMemory<char>& request_batch,
       std::shared_ptr<std::vector<TRITONBACKEND_Response*>>& responses);
+
+  // Get instance stub launcher
+  StubLauncher* Stub() { return instance_stub_; }
 };
 
+class StubLauncher {
+ public:
+  ~StubLauncher();
+
+  // Stub process setup
+  TRITONSERVER_Error* Setup(
+      ModelState* model_state, pid_t* parent_pid_, bool* initialized_,
```

Comment:
Can we make the Setup method as simple as possible and hide any unnecessary detail from the ModelState, ModelInstanceState? We can add accessor methods to the stub class to get the objects that ModelState, ModelInstanceState needs but I don't think we should make all the internal attributes of the stub launcher a pointer to the ModelState/ModelInstanceState attributes.

Reply:
> ...and hide any unnecessary detail from the ModelState, ModelInstanceState
Could you elaborate more on this? I'm not sure what unnecessary detail here means..

Reply:
I mean for example, I think the stub launcher doesn't need to have `parent_pid`, `model_state`, `initialized` etc as the setup arguments. These are internal fields to stub launcher that can be accessed using accessor methods (e.g. `Initialized()`, etc).

Reply:
I see. Thanks for the explanation. Will modify the Setup method.

Reply:
Thanks Kris!